plugin/storage/grpc/config/config.go

```diff
@@ -37,10 +42,34 @@ type Configuration struct {
 	RemoteTLS            tlscfg.Options
 	RemoteConnectTimeout time.Duration `yaml:"connection-timeout" mapstructure:"connection-timeout"`
 	TenancyOpts          tenancy.Options
+}
+
+type ConfigV2 struct {
+	Configuration
```

Comment:
You are still redefining v2 in terms of v1 - you need to break v1 dependency completely (super easy in case of this specific storage since v1 config is quire small). Otherwise you're just duplicating the settings, e.g. tenancy and TLS are also defined in v1 Configuration